runtimes/rust/src/api.rs

```diff
@@ -149,16 +149,23 @@ impl ServiceApiScope {
         Q: DeserializeOwned + 'static,
         I: Serialize + 'static,
         F: Fn(&ServiceApiState<'_>, Q) -> R + 'static + Clone + Send + Sync,
-        R: IntoFuture<Item = I, Error = crate::api::Error> + 'static,
+        R: IntoFuture<Item = I, Error = Error> + 'static,
     {
         let blockchain = self.blockchain.clone();
         let descriptor = self.descriptor.clone();
         self.inner
-            .endpoint(name, move |query: Q| -> crate::api::FutureResult<I> {
-                let descriptor = (descriptor.0, descriptor.1.as_ref());
-                let state = ServiceApiState::from_api_context(&blockchain, descriptor.into(), name);
+            .endpoint(name, move |query: Q| -> FutureResult<I> {
+                let (instance_id, instance_name) = descriptor.clone();
+                let state = ServiceApiState::from_api_context(
+                    &blockchain,
+                    (instance_id, instance_name.as_ref()).into(),
```

Comment:
I think it's better to bind this conversion to the variable. Maybe you'll be able to use `Display` representation of `InstanceDescriptor` for error source as well (not sure, since lifetimes can cause problems).